drivers/include/adt7310.h

```diff
@@ -0,0 +1,154 @@
+#define ADT7310_CONF_FAULT_QUEUE(x) (((x) << ADT7310_CONF_FAULT_QUEUE_SHIFT) & ADT7310_CONF_FAULT_QUEUE_MASK)
+#define ADT7310_CONF_CT_POL_MASK  (0x04)
+#define ADT7310_CONF_CT_POL_SHIFT (2)
+#define ADT7310_CONF_CT_POL(x) (((x) << ADT7310_CONF_CT_POL_SHIFT) & ADT7310_CONF_CT_POL_MASK)
+#define ADT7310_CONF_INT_POL_MASK  (0x08)
+#define ADT7310_CONF_INT_POL_SHIFT (3)
+#define ADT7310_CONF_INT_POL(x) (((x) << ADT7310_CONF_INT_POL_SHIFT) & ADT7310_CONF_INT_POL_MASK)
+#define ADT7310_CONF_INTCT_MODE_MASK  (0x10)
+#define ADT7310_CONF_INTCT_MODE_SHIFT (4)
+#define ADT7310_CONF_INTCT_MODE(x) (((x) << ADT7310_CONF_INTCT_MODE_SHIFT) & ADT7310_CONF_INTCT_MODE_MASK)
+#define ADT7310_CONF_OPERATION_MODE_MASK  (0x60)
+#define ADT7310_CONF_OPERATION_MODE_SHIFT (5)
+#define ADT7310_CONF_OPERATION_MODE(x) (((x) << ADT7310_CONF_OPERATION_MODE_SHIFT) & ADT7310_CONF_OPERATION_MODE_MASK)
+#define ADT7310_CONF_RESOLUTION_MASK  (0x80)
+#define ADT7310_CONF_RESOLUTION_SHIFT (7)
+#define ADT7310_CONF_RESOLUTION(x) (((x) << ADT7310_CONF_RESOLUTION_SHIFT) & ADT7310_CONF_RESOLUTION_MASK)
+
+/** @brief Continuous operation mode */
+#define ADT7310_MODE_CONTINUOUS (ADT7310_CONF_OPERATION_MODE(0))
+/** @brief One shot */
+#define ADT7310_MODE_ONE_SHOT   (ADT7310_CONF_OPERATION_MODE(1))
+/** @brief 1 sample per second */
+#define ADT7310_MODE_1SPS       (ADT7310_CONF_OPERATION_MODE(2))
+/** @brief Shut down (powersave) */
+#define ADT7310_MODE_SHUTDOWN   (ADT7310_CONF_OPERATION_MODE(3))
+/** @} */
+
+/**
+ * @brief Set configuration register of an ADT7310 sensor
+ *
+ * @param[in]  dev          device descriptor of sensor
+ * @param[in]  config       configuration byte, see macros in adt7310.h
+ *
+ * @return                  0 on success
+ * @return                  -1 on error
+ */
+int adt7310_set_config(adt7310_t *dev, uint8_t config);
+
+/**
+ * @brief Initialize the ADT7310 sensor driver.
+ *
+ * @param[out] dev          device descriptor of sensor to initialize
+ * @param[in]  spi          SPI bus the sensor is connected to
+ * @param[in]  cs           GPIO pin which the chip select signal is connected to
+ *
+ * @return                  0 on success
+ * @return                  <0 on error
+ */
+int adt7310_init(adt7310_t *dev, spi_t spi, gpio_t cs);
+
+/**
+ * @brief Read raw temperature register value
+ *
+ * @note The three least-significant bits of the value register are used for
+ *       flags if the sensor is configured for 13 bit mode.
+ *
+ * @param[in]  dev          device descriptor of sensor
+ * @param[out] raw          destination buffer
+ *
+ * @return                  0 on success
+ * @return                  -1 on error
+ */
+int adt7310_read(adt7310_t *dev, int16_t *raw);
+
+/**
+ * @brief Convert raw sensor values to temperature.
+ *
+ * @param[in]  raw          raw ADC value
+ *
+ * @return                  floating point representation of temperature in degrees Celsius
+ */
+float adt7310_convert(int16_t raw);
```

Comment:
good idea, however it will have to be `int32_t adt7310_read(dev);` in order to represent temperatures above 32.767 C